Next next invoke config

ronalag opened the pull request:
1. Modified the config parser to parse invoke targets.
2. Modified the native packager to generate targets in the native packager.
3. Updated the version of the packager tools.
4. Added unit tests for invoke target.

The issue related to this feature is:

blackberry/BB10-Webworks-Packager#111

The packager tools dependency had to be updated so that the native packager would recognize the invoke target elements in the descriptor and produce correct invoke target entries in the bar manifest file. The version of the updated native packager is 1.5.

If you want to grab the updated packager tools you can do so by grabbing the artifact of the ci build for this feature branch and copying the dependencies folder of the artifact to the project root directory.

Note: The updated packager tools does not support the signing of applications. Therefore this pull request should not be merged until signing is supported by the signing authority.

ronalag committed:
Made the following changes:
1. Modified the config parser to parse invoke targets.
2. Modified the native packager to generate targets in the native packager.
3. Updated the version of the packager tools.
4. Added unit tests for invoke target.

ronalag commented:
@EricLeiLi code review ?

tohman commented on lib/localize.js:
```diff
@@ -90,6 +90,12 @@ var Localize = require("localize"),
+            "en": "Each invokeTarget element must specify a valid invokeType and invokeId attribute"
+        },
+        "EXCEPTION_INVOKE_TARGET_FILTER_INVALID": {
+            "en": "Each filter element must specify at least one action and one type"
```
One action and one _mime_-type is more clear.

ronalag commented:
@EricLeiLi  The "- empty line + empty line" is there because I got rid of some extra whitespace in the native packager

ronalag committed:
Made the following changes:
1. Changed the wording of a configuration parsing error.
2. Got rid f unnecessary elements in the test configuration document for invoke targets.

ronalag committed:
Made the following changes:
1. Added comments to better explain the wrapPropertyInArray method.
2. Removed invoke target test config file and changed config parser unit tests to reference main config file.

ronalag commented:
@jkeshavarzi I made updates based on our discussion. Can you review them?

ronalag commented:
@jkeshavarzi the after each statement is necessary because some of the other tests parse different config.xml files in the test directory. Without this change some tests will be performed with the incorrect config.xml file. Having said that, I think that the after each should be a before each so I'll change the test case.

jamesjhedley commented:
The config.xml object should really be a contant and should not change from test to test, requiring us to reset it after or before each test. It should remain pointed to the standard config.xml for most of our tests. The other cases where a different config is used ("should" really only be one [bare-minimum.xml]) we can use local variables to store the different location.

----- Original Message -----
From: Henry Aghaulor [mailto:reply@reply.github.com]
Sent: Sunday, May 20, 2012 10:25 AM
To: James Keshavarzi
Subject: Re: [BB10-Webworks-Packager] Next next invoke config (#38)

@jkeshavarzi the after each statement is necessary because some of the other tests parse different config.xml files in the test directory. Without this change some tests will be performed with the incorrect config.xml file. Having said that, I think that the after each should be a before each so I'll change the test case.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/38#issuecomment-5809234

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

ronalag committed:
Modified test cases that change the config path in config parser

ronalag commented:
@jkesharvarshi made your recommended changes.

EricLeiLi commented:
@jkeshavarzi Thanks for helping the code review;
@ishneur please proceed to the SV&V.

jamesjhedley commented:
@EricLeiLi No problem.
@ronalag You should squash these commits before this branch is merged.

jeffheifetz commented:
@ronalag Is the spec you're coding to documented anywhere, its not present in the issue you referenced.

nukulb commented:
@jeffheifetz - it should on the wiki under invoke

nukulb commented:
@ronalag - rebase from next

ronalag commented:
@nukulb next or next-next?

nukulb commented:
@ronalag - next-next

ishneur commented:
For testing, is it sufficient to verify that correct information is added to blackberry-tablet.xml and to MANIFEST.MF, or should I also test the actual invocation of the WebWorks apps?

Also, I don't see a file called "bar-descriptor.xml". The file is called "blackberry-tablet.xml".

nukulb commented:
You actually need to test invocation

nukulb commented:
@ronalag - improve the commit message, please don't start all your commits with "made the follow changes" describe the change instead please. First line should always be a brief description.

ronalag committed:
Changed format of invoke target registration in config to match that of bar descriptor

ronalag committed:
Updated maven dependency to latest version
